std/assembly/process.ts

```diff
@@ -131,7 +131,7 @@ abstract class WritableStream extends Stream {
 
 @unmanaged
 abstract class ReadableStream extends Stream {
-  read(buffer: ArrayBuffer, offset: isize = 0): i32 {
+  read(buffer: ArrayBuffer, offset: usize = 0): i32 {
```

Comment:
With this change, the range check below, in particular `offset < 0`, will never be `true`. This can also be written differently of course, with an `usize` here, and explicit casts to `isize` below, if you think that's clearer?

Reply:
Where does it need to cast below?

Reply:
I'd assume you saw an error there, due to comparing an `isize` and an `usize` with `<`? If so, this can be fixed by keeping the code as-is, but changing the check below to `if (offset < 0 || <isize>offset > end)` I think. Or wasn't there a compilation error?

Reply:
I'm just confused why `offeset` is `isize` only to be used with `usize`'s.  For example, `<isize>offset > end` is what is causing the error in the original code since `end` is `usize`.  If there is motivation for this API then I can revert and just cast `offset` to `usize` -> `<usize>offset > end`.

Reply:
My expectation was that this API will be called with `i32`s most of the time, and upping that to `isize` was mostly an optimization to do the potential conversion exactly once when the function is called, instead of convert multiple times below. Other than in relational binary ops where there are two distinct Wasm ops to compare values (here: i32.lt_s and i32.lt_u), `isize` and `usize` are compatible.